jme3-core/src/main/java/com/jme3/scene/Node.java

```diff
@@ -249,6 +251,23 @@ public void updateGeometricState() {
         if ((refreshFlags & RF_LIGHTLIST) != 0) {
             updateWorldLightList();
         }
+
+        boolean updateGlobalLights = (refreshFlags & RF_GLOBAL_LIGHTS) != 0;
+        if (updateGlobalLights){
+            refreshFlags &= ~RF_GLOBAL_LIGHTS;
+            // if root node, we collect the global lights
+            if (getParent() == null){ 
+                depthFirstTraversal(sx->{
+                    LightList childLights = sx.getLocalLightList();
+                    for (Light l : childLights) {
+                        if (l.isGlobal()) {
+                            worldLights.add(l);
```

Comment:
I'm trying to think about ways to avoid this full traversal. Adding new lights is probably easy; only track down paths with RF_GLOBAL_LIGHTS looking for new lights. The getting rid of old lights is harder, I can't see anything in Light that records where a light came from (which would make removing old ones much easier)